packages/core/src/repositories/evaluationsRepository/index.ts

```diff
@@ -0,0 +1,176 @@
+import { omit } from 'lodash-es'
+
+import { and, eq, getTableColumns, inArray, isNull } from 'drizzle-orm'
+
+import { EvaluationDto } from '../../browser'
+import { EvaluationMetadataType } from '../../constants'
+import { NotFoundError, Ok, PromisedResult, Result } from '../../lib'
+import {
+  connectedEvaluations,
+  evaluationMetadataLlmAsJudgeCustom,
+  evaluationMetadataLlmAsJudgeLegacy,
+  evaluations,
+} from '../../schema'
+import { evaluationMetadataLlmAsJudgeBoolean } from '../../schema/models/evaluationMetadataLlmAsJudgeBoolean'
+import { evaluationMetadataLlmAsJudgeNumerical } from '../../schema/models/evaluationMetadataLlmAsJudgeNumerical'
+import { getSharedTableColumns } from '../../schema/schemaHelpers'
+import Repository from '../repository'
+
+const tt = {
+  ...getTableColumns(evaluations),
+  metadata: omit(
+    getSharedTableColumns(
+      evaluations.metadataType,
+      {
+        [EvaluationMetadataType.LlmAsJudgeLegacy]:
+          evaluationMetadataLlmAsJudgeLegacy,
+        [EvaluationMetadataType.LlmAsJudgeBoolean]:
+          evaluationMetadataLlmAsJudgeBoolean,
+        [EvaluationMetadataType.LlmAsJudgeNumerical]:
+          evaluationMetadataLlmAsJudgeNumerical,
+        [EvaluationMetadataType.LlmAsJudgeCustom]:
+          evaluationMetadataLlmAsJudgeCustom,
+      },
+      'metadata_columns',
+    ),
+    ['id', 'createdAt', 'updatedAt'],
+  ),
+}
+
+// const tt = {
+//   ...getTableColumns(evaluations),
+//   metadata: {
+//     id: sql`
+//       CASE
+//         WHEN ${evaluations.metadataType} = ${EvaluationMetadataType.LlmAsJudgeLegacy} THEN ${evaluationMetadataLlmAsJudgeLegacy.id}
+//         WHEN ${evaluations.metadataType} = ${EvaluationMetadataType.LlmAsJudgeBoolean} THEN ${evaluationMetadataLlmAsJudgeBoolean.id}
+//         WHEN ${evaluations.metadataType} = ${EvaluationMetadataType.LlmAsJudgeNumerical} THEN ${evaluationMetadataLlmAsJudgeNumerical.id}
+//       END
+//     `.as('id'),
+//     prompt: evaluationMetadataLlmAsJudgeLegacy.prompt,
+//     objective: sql`
+//       CASE
+//         WHEN ${evaluations.metadataType} = ${EvaluationMetadataType.LlmAsJudgeBoolean} THEN ${evaluationMetadataLlmAsJudgeBoolean.objective}
+//         WHEN ${evaluations.metadataType} = ${EvaluationMetadataType.LlmAsJudgeNumerical} THEN ${evaluationMetadataLlmAsJudgeNumerical.objective}
+//       END
+//     `.as('objective'),
+//     ...
+//   },
+// }
```

Comment:
The evaluation repository must return `EvaluationDto`, which now requires a bit of a complex query. Depending on the `evaluation.metadataType`, it will return different types for the `metadata`.

This new `getSharedTableColumns` function will handle this. It will generate a Drizzle query like the comment, where `metadata` contains all fields from all `MetadataTypes`, but get the data from a table depending on a `CASE` query.

Sadly, Drizzle cannot infer that this select will generate the actual `EvaluationDto` type, so we need to type it accordingly in the repository with an `as EvaluationDto` on every query.

All of this is tested on `evaluationsRepository/index.test.ts`, which shows that it works!

Reply:
(moved to future PR)